### ISSUE TITLE
local projects dependencies no longer work in gradle 6.6-rc-1

### ISSUE THREAD

**Issue by ZacSweers** [gradle/gradle#13811]
### Expected Behavior 
That they work

### Current Behavior

```
Caused by: org.gradle.internal.typeconversion.UnsupportedNotationException: Cannot convert the provided notation to a File or URI: full.jar (project :tooling:flipper:debug).
The following types/formats are supported:
  - A String or CharSequence path, for example 'src/main/java' or '/usr/include'.
  - A String or CharSequence URI, for example 'file:/usr/include'.
  - A File instance.
  - A Path instance.
  - A Directory instance.
  - A RegularFile instance.
  - A URI or URL instance.
  - A TextResource instance.
        at org.gradle.internal.typeconversion.ErrorHandlingNotationParser.parseNotation(ErrorHandlingNotationParser.java:57)
        at org.gradle.api.internal.file.AbstractFileResolver.convertObjectToFile(AbstractFileResolver.java:100)
        at org.gradle.api.internal.file.AbstractBaseDirFileResolver.doResolve(AbstractBaseDirFileResolver.java:39)
        at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:69)
        at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:48)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.visitSingleFile(UnpackingVisitor.java:105)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:100)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:91)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:91)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:83)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection$UnresolvedItemsCollector.visitContents(DefaultConfigurableFileCollection.java:355)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitChildren(DefaultConfigurableFileCollection.java:267)
        at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:108)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitDependencies(DefaultConfigurableFileCollection.java:273)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:112)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmptyEdges.getNodeValues(CachingDirectedGraphWalker.java:213)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:67)
        ... 107 more
```

### Context
We have a nightly job that tests configuration on newer gradle versions when they're available. Gradle 6.6 no longer works for our project with simple `project(...)` dependencies

In the above case, where the project `:tooling:flipper:debug` is an android kotlin library. I can trigger this on a different project as well with a plain java project (`:tooling:jsr250`) as the target dependency

```
Caused by: org.gradle.internal.typeconversion.UnsupportedNotationException: Cannot convert the provided notation to a File or URI: jsr250.jar (project :tooling:jsr250).
The following types/formats are supported:
  - A String or CharSequence path, for example 'src/main/java' or '/usr/include'.
  - A String or CharSequence URI, for example 'file:/usr/include'.
  - A File instance.
  - A Path instance.
  - A Directory instance.
  - A RegularFile instance.
  - A URI or URL instance.
  - A TextResource instance.
        at org.gradle.internal.typeconversion.ErrorHandlingNotationParser.parseNotation(ErrorHandlingNotationParser.java:57)
        at org.gradle.api.internal.file.AbstractFileResolver.convertObjectToFile(AbstractFileResolver.java:100)
        at org.gradle.api.internal.file.AbstractBaseDirFileResolver.doResolve(AbstractBaseDirFileResolver.java:39)
        at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:69)
        at org.gradle.api.internal.file.AbstractFileResolver.resolve(AbstractFileResolver.java:48)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.visitSingleFile(UnpackingVisitor.java:105)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:100)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:91)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:91)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:83)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection$UnresolvedItemsCollector.visitContents(DefaultConfigurableFileCollection.java:355)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitChildren(DefaultConfigurableFileCollection.java:267)
        at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:108)
        at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitDependencies(DefaultConfigurableFileCollection.java:273)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:112)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmptyEdges.getNodeValues(CachingDirectedGraphWalker.java:213)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:67)
        ... 107 more
```

in both cases, projects are depended on conventionally

```kotlin
dependencies {
  implementation(project(":tooling:flipper:debug"))
  implementation(project(":tooling:jsr250"))
}
```

### Steps to Reproduce 
Update to Gradle 6.6-rc-1 and use a local project dependency. Unfortunately our project is private

### Your Environment

Build scan URL: N/A

**Comment by runningcode** [comment]
These AGP issues may be relevant when investigating this issue:
https://issuetracker.google.com/issues/141126614
https://issuetracker.google.com/issues/158060799#comment2

**Comment by runningcode** [comment]
@ZacSweers what version of AGP are you using and what task were you running when encountering this issue?

**Comment by ZacSweers** [comment]
AGP 4.0.1. I wouldn't index on the lint task too much here, that just happens to be the task that we run on CI

**Comment by ZacSweers** [comment]
Well actually, it could be lint related. Not at my computer now but I do remember that we prepare our lint jars eagerly and possibly during sync. It does look familiar to this discussion in the Gradle community slack


https://gradle-community.slack.com/archives/CJYS1DAP5/p1594365315155400

**Comment by ZacSweers** [comment]
Fixed by #13812
